Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -970,9 +968,26 @@ private bool ExecuteOnAllServicesReverseOrder(Action<IMixedRealityService> execu
         private bool ExecuteOnAllServices(IEnumerable<IMixedRealityService> services, Action<IMixedRealityService> execute)
         {
             if (!HasProfileAndIsInitialized) { return false; }
+
+            // Get the count of services, so that we can protect against the count changing during iteration.
+            int serviceCount = -1;
+            // The MixedRealityServiceRegistry returns the service collection as an IReadOnlyCollection
+            IReadOnlyCollection<IMixedRealityService> serviceCollection = services as IReadOnlyCollection<IMixedRealityService>;
+            if (serviceCollection != null)
+            {
+                serviceCount = serviceCollection.Count();
+            }
+
             foreach (var system in services)
             {
                 execute(system);
+
+                // If we successfully got the count of services and the count has changed, exit the loop.
+                if ((serviceCount != -1) &&
```

Comment:
Why are we exiting early here? This seems like it's going to not execute services, potentiatlly randomly, if the services cound and MixedRealityServiceRegistry.GetAllServices().Count() do not match.

Reply:
foreach throws when the underlying collecction is changed (ex: when the profile swap happens to occur during this loop)